design/FY2017/NFP-Enhancement for Urban Microclimate.md

```diff
@@ -0,0 +1,99 @@
+
+![EplusUrban](EnergyPlusUrbanClimate.png)
+
+***Figure 1 EnergyPlus data exchange with urban microclimate model***
+
+The current implementation of EnergyPlus assumes a default vertical temperature gradient. It does not provide a direct link to allow data exchange with urban microclimate models to override local environmental data at the zone and surface levels, such as the outdoor air temperature, humidity, wind velocity and direction. However, there are reportable surface and zone outdoor conditions tracked separately as:
+
+	Zone,Average,Surface Outside Face Outdoor Air Drybulb Temperature [C]
+	Zone,Average,Surface Outside Face Outdoor Air Wetbulb Temperature [C]
+	Zone,Average,Surface Outside Face Outdoor Air Wind Speed [m/s]
+	Zone,Average,Zone Outdoor Air Drybulb Temperature [C]
+	Zone,Average,Zone Outdoor Air Wetbulb Temperature [C]
+	Zone,Average,Zone Outdoor Air Wind Speed [m/s]
+
+For a simple co-simulation case, it would be straightforward to add new EMS actuators for these six variables to override their values directly. We propose to add the Zone Outdoor Air Wind Direction [degree] as a new report variable at every time step. This would enable the co-simulation of the interaction between buildings and the urban micro-climate.
+
+
+# Approaches 
+We propose to add the Zone Outdoor Air Wind Direction [degree] report variable at every time step, and make the seven variables as new EMS actuators that can be set up for overriding the environmental variable fields. This enables data exchange with the local ambient air conditions at arbitrary nodes or coordinates by setting up FMUs for co-simulation (Figure 2).
+
+![LocalNode](EnergyPlusLocalAirNode.png) 
+
+***Figure 2 Local ambient air conditions at zone and surface level***
+
+The local environmental conditions can be exchanged with or overwritten by external simulation or calculation results at each time step in run time.
+
+# Testing/Validation/Data Sources
+
+TBD
+
+# Input Output Reference Documentation
+
+TBD
+
+# Engineering Reference
+
+TBD
+
+# Output Details and Examples
+
+TBD
+
+# Example Files and Transition Changes
+
+No transition changes are envisioned as a result of the new features (but this can change based on feedback from the team). New example files will be developed to demonstrate the new features.
+
+# References
+
+n/a
+
+# E-mail comments
+
+**Lixing Gu <gu@fsec.ucf.edu> Thu, Jun 1, 2017** 
+
+LocalOutdoorAir:Node
+Since you require to input outdoor conditions, T, W, Wind Speed and direction, is it possible to add solar radiation treated as diffuse component?
```

Comment:
Couldn't Gu's suggestion be extended to cover the external shading calculation feature proposal without requiring additional input objects?